### DISCUSSION TITLE
Craft Schema Version

### DISCUSSION THREAD

**Discussion by hiasl** [craftcms/cms#10192] [Ideas] [answered]
Currently Craft has a version and a schemaVersion in its config (info table).
As I have learned from Oli (Craft Team), the schemaVersion is increased whenever a new version of Craft brings migrations with it.

My Idea/Question: Wouldn't it be nice with to know how far back (in Craft versions) you could theoretically go with a dump? This could be done by setting the schemaVersion to the Craft version whenever migrations are applied in a certain Craft release.

Example (not taken from the current Craft releases): 
- Craft Version 3.7.22 is on schemaVersion 3.7.8. 
- Maybe this was the same for all Craft versions starting with version 3.7.15. 
- That means, 3.7.15 was the last Craft release bringing migrations with it. 
- In this case the schemaVersion could be 3.7.15 and not 3.7.8.

Now when dumps are created by `./craft db/backup` both the Craft Version and the Schema Version could become part of the filename, which makes identifying a certain dump much easier. It would tell you, that you could use any Craft version from 3.7.15 until now to work with that dump.

And maybe my idea is complete crap :-)

**Comment by brandonkelly** [comment, marked as answer]
There are some cases where multiple migrations are added to a single release, in which case the schema version is incremented multiple times. E.g. by the time 3.7.0 was tagged, the schema version was already up to 3.7.7:

https://github.com/craftcms/cms/blob/6948a8ab4b64a0224ba69cccb5c5a4deea196b19/src/config/app.php#L7

I like the idea though. Something to consider for 4.0 and onward. In cases where there are multiple migrations, we could just start incrementing another dot (4.0.0.10 instead of 4.0.10).

**Reply by brandonkelly** [reply]
Needed to bump the schema version for the upcoming 3.7.32 release, so went ahead and started applying this new convention ahead of schedule.
